editor/import/scene_import_settings.cpp

```diff
@@ -0,0 +1,1199 @@
+		}
+		if (selected_type == "Mesh" && mesh_map.has(selected_id)) {
+			const MeshData &md = mesh_map[selected_id];
+			rot_x = md.cam_rot_x;
+			rot_y = md.cam_rot_y;
+			zoom = md.cam_zoom;
+		} else if (selected_type == "Material" && material_map.has(selected_id)) {
+			const MaterialData &md = material_map[selected_id];
+			rot_x = md.cam_rot_x;
+			rot_y = md.cam_rot_y;
+			zoom = md.cam_zoom;
+		}
+	}
+
+	Vector3 center = camera_aabb.position + camera_aabb.size * 0.5;
+	float camera_size = camera_aabb.get_longest_axis_size();
+
+	camera->set_orthogonal(camera_size * zoom, 0.0001, camera_size * 2);
+
+	Transform xf;
+	xf.basis = Basis(Vector3(0, 1, 0), rot_y) * Basis(Vector3(1, 0, 0), rot_x);
+	xf.origin = center;
+	xf.translate(0, 0, camera_size);
+
+	camera->set_transform(xf);
+}
+
+void SceneImportSettings::_load_default_subresource_settings(Map<StringName, Variant> &settings, const String &p_type, const String &p_import_id, ResourceImporterScene::InternalImportCategory p_category) {
+	if (base_subresource_settings.has(p_type)) {
+		Dictionary d = base_subresource_settings[p_type];
+		if (d.has(p_import_id)) {
+			d = d[p_import_id];
+			List<ResourceImporterScene::ImportOption> options;
+			ResourceImporterScene::get_singleton()->get_internal_import_options(p_category, &options);
+			for (List<ResourceImporterScene::ImportOption>::Element *E = options.front(); E; E = E->next()) {
+				String key = E->get().option.name;
+				if (d.has(key)) {
+					settings[key] = d[key];
+				}
+			}
+		}
+	}
+}
+
+void SceneImportSettings::open_settings(const String &p_path) {
+	if (scene) {
+		memdelete(scene);
+		scene = nullptr;
+	}
+	scene = ResourceImporterScene::get_singleton()->pre_import(p_path);
+	if (scene == nullptr) {
+		EditorNode::get_singleton()->show_warning(TTR("Error opening scene"));
+		return;
+	}
+
+	base_path = p_path;
+
+	material_set.clear();
+	mesh_set.clear();
+	material_map.clear();
+	mesh_map.clear();
+	node_map.clear();
+	defaults.clear();
+
+	selected_id = "";
+	selected_type = "";
+
+	cam_rot_x = -Math_PI / 4;
+	cam_rot_y = -Math_PI / 4;
+	cam_zoom = 1;
+
+	{
+		base_subresource_settings.clear();
+
+		Ref<ConfigFile> config;
+		config.instance();
+		Error err = config->load(p_path + ".import");
+		if (err == OK) {
+			List<String> keys;
+			config->get_section_keys("params", &keys);
+			for (List<String>::Element *E = keys.front(); E; E = E->next()) {
+				Variant value = config->get_value("params", E->get());
+				if (E->get() == "_subresources") {
+					base_subresource_settings = value;
+				} else {
+					defaults[E->get()] = value;
+				}
+			}
+		}
+	}
+
+	first_aabb = true;
+
+	_update_scene();
+
+	base_viewport->add_child(scene);
+
+	if (first_aabb) {
+		contents_aabb = AABB(Vector3(-1, -1, -1), Vector3(2, 2, 2));
+		first_aabb = false;
+	}
+
+	popup_centered_ratio();
+	_update_camera();
+
+	set_title(vformat(TTR("Advanced import settings for '%s'"), base_path.get_file()));
```

Comment:
This title should be capitalized.
```suggestion
	set_title(vformat(TTR("Advanced Import Settings for '%s'"), base_path.get_file()));
```